File: src/main/java/de/meshcloud/example/springkotlin/config/DatabaseInitConfig.java
package de.meshcloud.example.springkotlin.config;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;

@Component
public class DatabaseInitConfig {

    @Value("${app.projectName}")
    private String projectName;

    public String getProjectName() {
        return projectName;
    }

    public void setProjectName(String projectName) {
        this.projectName = projectName;
    }
}

File: src/main/kotlin/de/meshcloud/example/springkotlin/config/DatabaseInitConfig.kt
package de.meshcloud.example.springkotlin.config

import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Component

@Component
class DatabaseInitConfig(

    // prefer @Value over @ConfigurationProperties, because otherwise variables have to be "lateinit var"
    // and therefore mandatory config properties won't cause the application fail during start, but only throw
    // an exception during runtime
    @Value("\${app.projectName}")
    val projectName: String
)